compiler/cpp/src/thrift/generate/t_c_glib_generator.cc

```diff
@@ -4061,8 +4061,17 @@ void t_c_glib_generator::generate_deserialize_field(ostream& out,
       throw "compiler error: no C reader for base type " + t_base_type::t_base_name(tbase) + name;
     }
     out << ", error)) < 0)" << endl;
-    out << indent() << "  return " << error_ret << ";" << endl << indent() << "xfer += ret;"
-        << endl;
+    if ((tbase != t_base_type::TYPE_STRING) && allocate) {
```

Comment:
Why is that test different to line 4022 (allocate is not tested there)? Shouldn't that be consistent?

Reply:
4022 is :
![image](https://user-images.githubusercontent.com/34181886/114119965-34de8900-991e-11eb-97e2-28b18f3a02ef.png)
If it's not equal t_base_type::TYPE_STRING , I've increased the processing。The two are different,and I think it's an omission.